libbpf-tools/doublefree.c

```diff
@@ -0,0 +1,594 @@
+/* SPDX-License-Identifier: BSD-2-Clause */
+/* Copyright (c) 2022 LG Electronics */
+#include <argp.h>
+#include <stdio.h>
+#include <signal.h>
+#include <stdarg.h>
+#include <stdlib.h>
+#include <string.h>
+#include <unistd.h>
+#include <bpf/libbpf.h>
+#include <bpf/bpf.h>
+#include "doublefree.h"
+#include "doublefree.skel.h"
+#include "trace_helpers.h"
+#include "uprobe_helpers.h"
+
+enum log_level {
+	DEBUG,
+	INFO,
+	WARN,
+	ERROR,
+};
+
+void __p(enum log_level level, char *level_str, char *fmt, ...);
+void set_log_level(enum log_level level);
+
+#define p_debug(fmt, ...) __p(DEBUG, "Debug", fmt, ##__VA_ARGS__)
+#define p_info(fmt, ...) __p(INFO, "Info", fmt, ##__VA_ARGS__)
+#define p_warn(fmt, ...) __p(WARN, "Warn", fmt, ##__VA_ARGS__)
+#define p_err(fmt, ...) __p(ERROR, "Error", fmt, ##__VA_ARGS__)
+
+pid_t execute_process(char *cmd);
+
+#define ON_MEM_FAILURE(buf) \
+if (NULL == buf) { \
+	p_err("FATAL: Failed to allocate memory on %s", __func__); \
+	exit(-1); \
+}
+
+/* default log level */
+static enum log_level log_level = ERROR;
+
+void sig_handler(int signo)
```

Comment:
static ? Here and below.

Reply:
There is a reason why I use static here and below. (below means static functions, right?)
I could remove static but I'd like to use static in here and below, if there is no problem.

I use this tool and other tools in my company, and manage the source code bit differently.
I compile every tool into one binary for user to use easily. So symbol can conflict on compilation if I don't use static.

Reply:
I removed static on this enum value.